src/Utilities/api.js

```diff
@@ -0,0 +1,95 @@
+class Geocoder {
+  constructor(apiKey) {
+    /*
+    Makes geocoder api instances for connecting to a geocoding api with the provided credentials..
+    Args: apiKey (str) - api key to use for requests
+    Return: Geocoder instance (obj)
+    */
+    this.key = apiKey;
+    this.endpoints = {
+      forwardGeocode: 'https://us1.locationiq.com/v1/search.php?key=',
+      reverseGeocode: 'https://us1.locationiq.com/v1/reverse.php?key='
+    };
+  }
+
+  _checkResponseJSON(response) {
```

Comment:
is the _ needed? javascript doesnt really follow that practice anymore and can be confusing to other developers without comments but if it is not needed then best to not use it.

Reply:
Oh so I don't have to mark those kinds of internally used methods for a class?

Reply:
Refactored it like 7 times to make it more of a functional style. I went down a real rabbit hole rewriting this and with all the further reading I had to do.